plugins/woocommerce/client/legacy/css/admin.scss

```diff
@@ -8019,3 +8019,52 @@ table.bar_chart {
 		width: calc(100% - 24px);
 	}
 }
+
+.woocommerce-contextual-notice {
+	margin-top: 1em;
+	background-color: lightgray;
+	padding: 12px 16px;
+
+	.woocommerce-contextual-notice-cta {
+		padding-top: 1em;
+	}
+
+	// Default to "info" theme.
+	background-color: #F0F6FC;
+}
+
+.woocommerce-contextual-notice-success {
+	// These are shared / Gutenberg colours, replace with variables :)
+	background-color: #f0fcf0;
+
+	.woocommerce-contextual-notice-cta .components-button,
+	.woocommerce-contextual-notice-cta .components-button:hover:not(:disabled) {
+		box-shadow: inset 0 0 0 1px #18cc3f;
+		border-color: #18cc3f;
+		color: #18cc3f;
+	}
+}
+
+.woocommerce-contextual-notice-warning {
+	// These are shared / Gutenberg colours, replace with variables :)
+	background-color: #FCF9E8;
+
+	.woocommerce-contextual-notice-cta .components-button,
+	.woocommerce-contextual-notice-cta .components-button:hover:not(:disabled) {
+		box-shadow: inset 0 0 0 1px #BD8600;
+		border-color: #BD8600;
+		color: #BD8600;
+	}
+}
+
+.woocommerce-contextual-notice-error {
+	// These are shared / Gutenberg colours, replace with variables :)
+	background-color: #FCF0F1;
+
+	.woocommerce-contextual-notice-cta .components-button,
+	.woocommerce-contextual-notice-cta .components-button:hover:not(:disabled) {
+		box-shadow: inset 0 0 0 1px #CC1818;
+		border-color: #CC1818;
+		color: #CC1818;
+	}
```

Comment:
This CSS is all working for my purposes, but I'm keen to refine it a lot more, with guidance from experts & testing from reviewers. Also design folks could have useful input here.

1. The colours used are all official / Woo / Gutenberg colours. Do we have scss variables I can use instead of pasting raw hex?
2. I've overridden existing `.components-button` (to change colours). Need to stress test this and ensure there aren't any hover/active etc states which are showing wrong (default) colours.
3. Class naming – I'm keen for this to be a reusable component, and to name the classes consistent with however we do that. e.g. `.component-contextual-notice` etc.
4. Component naming - I've named this `woocommerce-contextual-notice`; keen for ideas for the right name for this. Could be semantic – name for specific purpose (i.e. very important, non-dismissable notices in context); or low-level – name as raw elements that can be used for any purpose.
5. Padding / margins – I've hacked to get the layout I need. Might need some more thought, care, testing to get this right for general use (e.g. in other contexts).

Reply:
I notice that the disputes project Figma design (RiEQmaKRI7u9USAI3lyZz0VX-fi-2885_194324) the component Inline Notice from the WC Admin design library: Kofpo8jj4zDWXDsVdlSz0f-fi-4254_46957. Does this component and associated styles already exist within the WC codebase?

Reply:
> Based on this, should we consider the naming woocommerce-inline-notice?

> Does this component and associated styles already exist within the WC codebase?

This is similar to @louwie17  [feedback](https://github.com/woocommerce/woocommerce/pull/38443#pullrequestreview-1455343676) 😁 

> We also encourage the use of the React notice component from Gutenberg, and as more and more of the UI is being replaced by React, using the notice component would make more sense.

I'm going to rework to just add the hook and then use the Gutenberg / React notice component to render a nice notice inline.